CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/gui/VisualGraphTopComponent.java

```diff
@@ -1093,6 +1095,29 @@ protected Void doInBackground() throws Exception {
                 rg.release();
             }
             try {
+                
+                // Create a 'backup' copy of the file being saved before its saved. This is done to ensure that there are
+                // two distinct saves/write operations that occur meaning that if the application was to terminate on one
+                // of them, then 'one' of the files should be valid still.
+                // i.e.:
+                //  *  if the 'copy' operation fails, then the original file will still be there intact
+                //  *  if the 'write' operation falls over, then we know the backup copy must have already occured so we
+                //     have something to fall back on
+                // This allows load logic to be of the form:
+                //   - try to load star file
+                //     - if star file loads successfully use it
+                //     - if the star file fails to load, check to see if a 'backup' exists
+                //        - if backup was found attempt to load it
+                //        - if backup was not found throw load error
+                FileObject fileobj = freshGdo.getPrimaryFile();
+                File srcFile = new File(fileobj.getPath());
+                String srcfilePath = srcFile.getParent().concat("\\").concat(this.name).concat(".").concat(fileobj.getExt());
```

Comment:
Should the `\\` be replaced with `File.pathSeparator` so we support Linux and Mac?

Reply:
`SeparatorConstants` might have a `.PERIOD` also

Reply:
done - for path seperatore .. .. left the . as its universal.